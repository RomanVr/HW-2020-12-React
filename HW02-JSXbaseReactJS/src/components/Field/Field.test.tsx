import React from "react";
import { shallow } from "enzyme";
import { Field } from "..";

describe("Field render", () => {
  it("Render Field without props", () => {
    const component = shallow(<Field />);
    expect(component).toMatchSnapshot();
  });

  describe("Render 3x3 Field contain Cell-component is matrix", () => {
    const size = 3;
    const component = shallow(<Field start={size}></Field>);
    const wrapperCellAll = component.find("Cell");

    it("Render Field contain Cell-component should be 9", () => {
      expect(wrapperCellAll.length).toBe(size * size);
    });

    it("Render Field contain last Cell-component should be key is 8", () => {
      const wrapperCellLast = wrapperCellAll.last();
      expect(wrapperCellLast.prop("dataKey")).toBe(size * size - 1);
    });
  });
});
