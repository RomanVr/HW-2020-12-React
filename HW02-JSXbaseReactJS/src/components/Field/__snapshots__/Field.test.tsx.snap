// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field render Render Field without props 1`] = `ShallowWrapper {}`;
